R/iterate-responses.R

```diff
@@ -0,0 +1,34 @@
+#' Tools for working with lists of responses
+#'
+#' * `resps_combine()` combines the data from each responses into a single
+#'   object.
+#' * `resps_response()` returns all successful responses.
+#' * `resps_error()` returns all errors.
+#'
+#' @export
+#' @param resps A list of responses (possibly including errors).
+#' @param resp_data A function that takes a response (`resp`) and
+#'   returns its data as a vector or data frame.
+#' @param resp_data A function with one argument `resp` that parses the
+#'   response and returns a list with the field `data` and other fields needed
+#'   to create the request for the next page.
+resps_combine <- function(resps, resp_data) {
+  check_installed("vctrs")
+
+  check_function2(resp_data, "body")
```

Comment:
I'm pretty sure the argument (also as documented above) is `resp`, not `body`

```suggestion
  check_function2(resp_data, "resp")
```